### PR TITLE
Feature/save sessions to db

### PR DESCRIPTION
**Pull request by tefrati**
Sessions are persisted in DB
Sessions DB and App DB are handled in 2 separate files: one in fbm submodule and the other in the app level. Firebase app is initialized in submodule.
apiai contexts and app data are stored in separate properties in session
created two common methods for all channels: getUserProfile and startChannel

### CODE REVIEW

File: sessionsManager.js [hunk 1]
```diff
@@ -329,7 +328,7 @@ module.exports.SOURCE_TYPE = SOURCE_TYPE;
 module.exports.CHANNELS = CHANNELS;
 module.exports.handleEventBySessionId = handleEventBySessionId;
 module.exports.handleEventByUserChannelId = handleEventByUserChannelId;
-module.exports.getSessionContext = getSessionContext;
-module.exports.initializeDb = initializeDb;
```
Comment (tbass134) [inline sessionsManager.js]: Breaking change should be documented in pull request

File: channels/facebook/wphook.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ const rpn = require('request-promise-native')
 const WORKPLACE_VERIFY_TOKEN = process.env.WORKPLACE_VERIFY_TOKEN;
 const WORKPLACE_PAGE_ACCESS_TOKEN = process.env.WORKPLACE_PAGE_ACCESS_TOKEN
 
+var startChannel= () => {
```
Comment (tbass134) [inline channels/facebook/wphook.js]: is this function neccessary?

File: sessionsManager.js [hunk 3]
```diff
@@ -26,9 +26,9 @@ const MESSAGE_TYPES = {
 };
 
 const CHANNELS = {
-	FB_MESSENGER: Symbol("FB_MESSENGER"),
```
Comment (tbass134) [inline sessionsManager.js]: 🥇 👍

File: sessionsManager.js [hunk 4]
```diff
@@ -120,68 +126,44 @@ var getSessionByChannelEvent = (messagingEvent) => {
...
-			}
-			getUserProfilePromise
+			getChannel(mappedChatSession.channelType).getUserProfile(mappedChatSession.source)
 			.then(json => {
 				console.log("user profile:" + JSON.stringify(json));
 				mappedChatSession.profile = json;
 				return mappedChatSession;
 			})
 			.then(session => {
 				sessionsDb.saveSession(session)
```
Comment (tbass134) [inline sessionsManager.js]: doesn't saveSession return a promise?

File: sessionsManager.js [hunk 5]
```diff
@@ -172,15 +172,21 @@ var getSessionByChannelEvent = (messagingEvent) => {
...
+			delete userChannelToSessions[source]
+			delete chatSessions[session.sessionId]
+			sessionsDb.removeSession(session.sessionId)
+			.then(sessionId => {
+				resolve(sessionId)
+			})
+		}
+		else {
+			console.log("removeSessionBySource: no session was found for source: " + source)
+			resolve(-1)
```
Comment (tbass134) [inline sessionsManager.js]: shouldn't this return reject() instead of resolve()?

File: sessionsManager.js (hunk 1 above)
Comment (tefrati) [reply sessionsManager.js]: Agreed

File: channels/facebook/wphook.js (hunk 2 above)
Comment (tefrati) [reply channels/facebook/wphook.js]: No and yes:
For WP and FBM we don't need to. However, I'm trying to conceptualize common methods across all channels. startChannel is used today with Nexmo, and I suspect that in the future we would to som something with WP/FBM as well

File: sessionsManager.js (hunk 5 above)
Comment (tefrati) [reply sessionsManager.js]: it's a question of preference. I'm not married to any particular approach

**Pull request merged**